### ISSUE TITLE
QuickLook crashes after Windows 10 20H2 upgrade

### ISSUE THREAD

**Issue by eric3272** [QL-Win/QuickLook#948]
Ever since the 20H2 upgrade, QuickLook crashes every single time I touch the space bar. I installed it from Windows Store, version 3.6.11.  Also tried installing it using the msi file, still seeing exactly the same issue. I'm not 100% sure if this is related to the Windows upgrade, but could someone please let me know what I can do to have it fixed? 

Edit: 

Describe the bug
The crashes start after getting the Windows 10 update KB5003537 on Windows 10 20H2 and trying to preview a large JSON file(over 25MB), then QuickLook becomes unresponsive and has to be force quit. After the force quit, QuickLook will crash immediately every time no matter what file type I want to preview.

To Reproduce

Get the Windows 10 update KB5003537 on Windows 10 20H2
Preview a large JSON file (over 25MB) and QuickLook becomes unresponsive
Force quit QuickLook when Windows 10 prompts
Reopen QuickLook and try to preview any file, QuickLook will crash.
Expected behavior
QuickLook will keep working as normal.

Desktop (please complete the following information):

OS Version: Windows 10 Pro 20H2, OS build 19042.1081
QuickLook Version: 3.6.11 both from MS store and the msi installation package

Screenshots and Exception Log
![image](https://user-images.githubusercontent.com/63749851/124513026-3221f180-ddb0-11eb-95ab-799a08b49321.png)

**Comment by eric3272** [comment]
After checking the Event Viewer logs, I found that the problem might have been caused by a windows 10 update (KB5003537), uninstalled the update and QuickLook is working again. 

Attached below is the log content

Faulting application name: QuickLook.exe, version: 3.6.11.0, time stamp: 0x60451623
Faulting module name: clr.dll, version: 4.8.4390.0, time stamp: 0x609c45bd
Exception code: 0xc0000409
Fault offset: 0x0000000000332a8c
Faulting process id: 0x1a8c
Faulting application start time: 0x01d76e8f707b4a35
Faulting application path: C:\Program Files\WindowsApps\21090PaddyXu.QuickLook_3.6.11.0_neutral__egxr34yet59cg\Package\QuickLook.exe
Faulting module path: C:\Windows\Microsoft.NET\Framework64\v4.0.30319\clr.dll
Report Id: 10b67ff3-8e3c-48d1-99f4-8d681ac5050e
Faulting package full name: 21090PaddyXu.QuickLook_3.6.11.0_neutral__egxr34yet59cg
Faulting package-relative application ID: Main

**Comment by eric3272** [comment]
An update on this issue. 

I reinstalled the Windows 10 update KB5003537 once QuickLook started working again after I had uninstalled it, then QuickLook was working fine with the Windows 10 update installed until I tried to preview a large JSON(over 20MB) and QuickLook became unresponsive, so I had to force quit it. And this is when it started to crash again every time I try to use it, had to uninstall the windows 10 update again to have QuickLook back. I'm seeing the exact same Event Viewer logs for this error.

**Comment by xupefei** [comment]
Could you modify the issue [according to the template](https://github.com/QL-Win/QuickLook/issues/new?assignees=&labels=&template=bug_report.md)? Only so I know what exactly is the problem and how to reproduce it.

**Comment by eric3272** [comment]
**Describe the bug**
The crashes start after getting the Windows 10 update KB5003537 on Windows 10 20H2 and trying to preview a large JSON file(over 25MB), then QuickLook becomes unresponsive and has to be force quit. After the force quit, QuickLook will crash immediately every time no matter what file type I want to preview. 

**To Reproduce**
Steps to reproduce the behavior:
1. Get the Windows 10 update KB5003537 on Windows 10 20H2
2. Preview a large JSON file (over 25MB) and QuickLook becomes unresponsive
3. Force quit QuickLook when Windows 10 prompts
4. Reopen QuickLook and try to preview any file, QuickLook will crash.

**Expected behavior**
QuickLook will keep working as normal.

**Desktop (please complete the following information):**
 - OS Version: Windows 10 Pro 20H2, OS build 19042.1081
 - QuickLook Version: 3.6.11 both from MS store and the msi installation package

**Screenshots and Exception Log**
![image](https://user-images.githubusercontent.com/63749851/124512865-d192b480-ddaf-11eb-9c5a-81e87770c3e4.png)

**Comment by xupefei** [comment]
Mind to share me the 20MB JSON file?

**Comment by eric3272** [comment]
Yes, I have thousands of them and I don't remember which one it was that I tried to preview that caused the crashes, but I believe all of them should do as long as it causes QuickLook to freeze due to the relatively large file size. I'm attaching one of them and hope it'll help. 

[All-high.zip](https://github.com/QL-Win/QuickLook/files/6770639/All-high.zip)

**Comment by eric3272** [comment]
Not sure if this will help, but I have Atom as the default app to open JSON files.

**Comment by xupefei** [comment]
I see a large amount of time spent in Regex matching when trying to apply syntax highlight to the JSON file.
Can you try https://ci.appveyor.com/project/xupefei/quicklook/builds/39890560/artifacts and see if the issue has been fixed?

**Comment by eric3272** [comment]
Hi Paddy, I'm still experiencing the same issue with the new MSI, attached is the new Event Viewer log screenshot. 

I reinstalled the update KB5003537 and was paying attention to not open large JSON with QuickLook, so it was working fine. But after uninstalling the MS store version and unzipped the folder to the QuickLook folder, then tried to open a large JSON, this time it skipped the steps of QuickLook becoming unresponsive and having to be force quit, it's now crashing every time I try to preview any file. 

![image](https://user-images.githubusercontent.com/63749851/124657256-aaec8080-de78-11eb-830c-4448a11f9af8.png)

**Comment by xupefei** [comment]
That's weird. Could you reboot and try again?
Also, do you have a crash log at any of these locations: https://github.com/QL-Win/QuickLook/wiki/Differences-Between-Distributions#user-data-location?

**Comment by eric3272** [comment]
Hey Paddy, thanks for the follow-up! I have to run, will update you when I'm back or tomorrow!

**Comment by eric3272** [comment]
Hi, I installed 3.6.11-21 with the msi package and it's now working fine, I will update you once anything comes up again. 

I didn't find logs from any of the listed locations in the link.

Also, I realized that the whole issue was probably not caused by the Windows 10 update KB5003537, because the MS store version QuickLook started to work again for a few days after I uninstalled the update, then I started to have the exact same issue with QuickLook (MS store 3.6.11) crashing every time I used it, and the Event Viewer log states the same issue with the clr.dll file.

I'll come back if anything goes wrong again. Thank you very much for the help!

**Comment by xupefei** [comment]
Wonderful. Please reopen this ticket if you see the same issue again.

**Comment by eric3272** [comment]
Hey Paddy,

I'm sorry that the same issue is happening again even though I'm currently using the 3.6.11-21 version, this time it seems to have started with another Windows 10 update on July 11, I uninstalled the update and QuickLook is now working again. Event Viewer logs suggest it's the same clr.dll file that's the faulting module. I checked the listed locations for crash logs, but didn't find anything. However, I do have a few crash dump files and I'm attaching the latest one and hope that would help. Let me know if you would need any other information from me. 


[QuickLook.exe.18244.zip](https://github.com/QL-Win/QuickLook/files/6797332/QuickLook.exe.18244.zip)

**Comment by xupefei** [comment]
How about deleting the TextViewer plugin?

**Comment by eric3272** [comment]
Thanks for the response Paddy, I'll try as soon as I get the chance and let you know what happens.

**Comment by eric3272** [comment]
Hi Paddy, 

I tried deleting the TextViewer plugin, but unfortunately, I'm still having the exact same issue. I found another Windows Error Report and the screenshot of it is attached. I don't know if the "Report.wer" file would be of any value to you, but if it's necessary to you, it's attached in the zip file. 

Also, the issue may not be related to Windows updates because I paused Windows updates since the last time I experienced this issue. But QuickLook 3.6.11-21 all of a sudden started to crash again today even without any Windows updates installed. Please let me know how I can help and I'll share anything with you. 


[Report.zip](https://github.com/QL-Win/QuickLook/files/6825243/Report.zip)
![image](https://user-images.githubusercontent.com/63749851/125833298-c42b2fb7-d5b5-41b4-ab12-4d438394f0e7.png)

**Comment by eric3272** [comment]
Just an update, I restarted my computer after deleting the TextViewer plugin, and QuickLook is now working for most other file types, when I put the TextViewer plugin back in the QuickLook.Plugin folder, it started to crash again, so I guess it might be related to the TextViewer plugin?

**Comment by Roundlay** [comment]
> Just an update, I restarted my computer after deleting the TextViewer plugin, and QuickLook is now working for most other file types, when I put the TextViewer plugin back in the QuickLook.Plugin folder, it started to crash again, so I guess it might be related to the TextViewer plugin?

This was really helpful to know. I was consistently encountering a hang and subsequent crash on my first attempt to run QuickLook, which would curiously resolve itself in subsequent runs. After I removed the TextViewer plugin, I noticed an immediate improvement. The preview times are remarkably faster and the previous issues seem to have disappeared.
